PolyDock/PolyDock/src/pd/ecs/cmp/tabsHeader/TabsAdding.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <optional>
+#include <entt/entt.hpp>
+
+#include <pd/WindowTabHandle.hpp>
+
+namespace pd::ecs::cmp::tabsHeader
+{
+	// ---------------------------------------------------------------------------------------------------------
+	class AddButtonHovered
+	{
+	};
+
+	// ---------------------------------------------------------------------------------------------------------
+	class AddButtonPressed
+	{
+	};
+
+	// ---------------------------------------------------------------------------------------------------------
+	class AddTabRequest
+	{
+	public:
+		class Request
+		{
+		public:
+			bool focus;
+			std::optional<int> index;
+			std::optional<entt::entity> newTabEntity;
+		};
+
+		std::vector<Request> requests;
```

Comment:
When do we need multiple requests? For example restoring old tabs?

Reply:
We need in case if user or other system want to add multiple tabs within single update.